service/src/main/java/uk/gov/hmcts/reform/fpl/service/ApplicantService.java

```diff
@@ -3,37 +3,73 @@
+import uk.gov.hmcts.reform.rd.model.Organisation;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
+import static java.util.Objects.isNull;
 import static java.util.stream.Collectors.toList;
 import static org.springframework.util.ObjectUtils.isEmpty;
 
 @Service
 public class ApplicantService {
 
-    public List<Element<Applicant>> expandApplicantCollection(CaseData caseData) {
+    public List<Element<Applicant>> expandApplicantCollection(CaseData caseData, Organisation organisation) {
         if (isEmpty(caseData.getApplicants())) {
-            return ImmutableList.of(Element.<Applicant>builder()
-                .value(Applicant.builder()
-                    .party(ApplicantParty.builder()
-                        // A value within applicant party needs to be set in order to expand UI view.
-                        .partyId(UUID.randomUUID().toString())
+            if (isEmpty(organisation)) {
+                return ImmutableList.of(Element.<Applicant>builder()
+                    .value(Applicant.builder()
+                        .party(ApplicantParty.builder()
+                            // A value within applicant party needs to be set in order to expand UI view.
+                            .partyId(UUID.randomUUID().toString())
+                            .build())
                         .build())
-                    .build())
-                .build());
+                    .build());
+            }
+            return buildApplicantWithOrganisationDetails(organisation);
 
         } else {
             return caseData.getApplicants();
         }
     }
 
+    private List<Element<Applicant>> buildApplicantWithOrganisationDetails(Organisation organisation) {
+        ContactInformation contactInformation = ContactInformation.builder().build();
+
+        if (!isNull(organisation.getContactInformation())) {
+            contactInformation = organisation.getContactInformation().get(0);
+        }
+
+        return ImmutableList.of(Element.<Applicant>builder()
+            .value(Applicant.builder()
+                .party(ApplicantParty.builder()
+                    .organisationName(organisation.getName())
+                    .address(buildApplicantAddressWithOrganisationDetails(contactInformation))
+                    .build())
+                .build())
+            .build());
+    }
+
+    private Address buildApplicantAddressWithOrganisationDetails(ContactInformation contactInformation) {
+        return Address.builder()
+            .addressLine1(contactInformation.getAddressLine1())
+            .addressLine2(contactInformation.getAddressLine2())
+            .addressLine3(contactInformation.getAddressLine3())
+            .county(contactInformation.getCounty())
+            .country(contactInformation.getCountry())
+            .postcode(contactInformation.getPostCode())
+            .postTown(contactInformation.getTownCity())
+            .build();
+    }
```

Comment:
can we not create method in ContactInformation called toAddress or buildAddress, which returns Address object?

Reply:
good idea, done !